### PR TITLE
(docs) - Add Comparison page

### PR DESCRIPTION
**Pull request by kitten**
This aims to add an unbiased and comprehensive comparison page that can be expanded in the future as we see fit. I'm trying to keep it accurate but I expect that it'll change as more questions arise or other libraries change. For now we only compare `urql` with Apollo and Relay and not other libraries like React Query or SWR as we only want to have direct comparisons with GraphQL clients... well, for just GraphQL. Otherwise the feature sets wouldn't make much sense.

### CODE REVIEW

File: docs/comparison.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+| Retrying Failed Queries                    | ✅ `@urql/exchange-retry`           | ✅ `apollo-link-retry`                        | ✅ `DefaultNetworkLayer`       |
+| Easy Authentication Flows                  | ✅ `@urql/exchange-auth`            | 🛑 (no docs for refresh-based authentication) | 🟡 `react-relay-network-layer` |
+| Automatic Refetch after Mutation           | ✅ (with document cache)            | 🛑                                            | ✅                             |
+
+Typically these are all additional addon features that you may expect from a GraphQL client, no
+matter which framework you use it with. It's worth mentioning that all three clients support some
+kind of extensibility API which allows you to change when and how queries are sent to an API. These
+are easy to use primitives particularly in Apollo, with links, and in `urql` with exchanges. The
+major difference in `urql` is that all caching logic is abstracted in exchanges too, which makes
+them slightly more powerful (and hence makes `urql` slightly more customizable.)
```
Comment (JoviDeCroock) [inline docs/comparison.md:63]: I'd refrain from using powerful and just stick to "more customizable"

File: docs/comparison.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+A lot of the added exchanges for persisted queries, file uploads, retrying, and other features are
+implemented in the core library, while there are some cases where first-party support isn't provided
```
Comment (JoviDeCroock) [inline docs/comparison.md:66]:
```suggestion
implemented by the urql-team, while there are some cases where first-party support isn't provided
```

It's not really part of the core-library

File: docs/comparison.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+answered quickly enough on [GitHub Discussions](https://github.com/FormidableLabs/urql/discussions)
+for instance.
+
+### Caching and State
+
+| ---                                                     | urql                                                                  | Apollo             | Relay                                          |
+| ------------------------------------------------------- | --------------------------------------------------------------------- | ------------------ | ---------------------------------------------- |
+| Caching Strategy                                        | Document Caching, Normalized Caching with `@urql/exchange-graphcache` | Normalized Caching | Normalized Caching (schema restrictions apply) |
+| Added Bundle Size                                       | +5.9kB (with Graphcache)                                              | +0 (default)       | +0 (default)                                   |
+| Automatic Garbage Collection                            | ✅                                                                    | ✅                 | ✅                                             |
```
Comment (JoviDeCroock) [inline docs/comparison.md:99]: Is it automatic in apollo, thought you had to call .gc

File: docs/comparison.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+| Optimistic Updates                                      | ✅                                                                    | ✅                 | ✅                                             |
+| Local Updates                                           | ✅                                                                    | ✅                 | ✅                                             |
+| Out-of-band Cache Updates                               | 🛑 (stays true to server data)                                        | ✅                 | 🛑                                             |
+| Local Resolvers and Redirects                           | ✅                                                                    | ✅                 | 🛑                                             |
+| Complex Resolvers (nested non-normalized return values) | ✅                                                                    | 🛑                 | 🛑                                             |
+| Commutativity Guarantees                                | ✅                                                                    | 🛑                 | 🛑                                             |
```
Comment (JoviDeCroock) [inline docs/comparison.md:107]: Could link to under-the-hood to explain this

File: docs/comparison.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+| Partial Results                                         | ✅                                                                    | ✅                 | 🛑                                             |
+| Partial Results                                         | ✅                                                                    | ✅                 | 🛑                                             |
```
Comment (JoviDeCroock) [inline docs/comparison.md:109]: Partial results is duplicated her

File: docs/comparison.md (hunk 3 above)
Comment (kitten) [reply docs/comparison.md:99]: That's true, I just checked and it _isn't_ always automatic, I suppose to not break legacy apps, so I'll update this

File: docs/comparison.md (hunk 4 above)
Comment (kitten) [reply docs/comparison.md:107]: It's explained in the explanation after the table

File: docs/comparison.md [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+customizable.)
+
+A lot of the added exchanges for persisted queries, file uploads, retrying, and other features are
+implemented by the urql-team, while there are some cases where first-party support isn't provided
+in Relay or Apollo. This doesn't mean that these features can't be used with these clients, but that
+you'd have to lean on community libraries or maintaining/implementing them yourself.
+
+### Framework Bindings
+
+| ---                           | urql                             | Apollo             | Relay             |
```
Comment (stevenmusumeche) [inline docs/comparison.md:73]: the order of the columns changed - maybe keep them consistent?

Comment (stevenmusumeche) [reply docs/comparison.md:73]: you're right - wrapping in the github UI

**Review by andyrichardson — APPROVED** [review]

**Pull request merged**